File: app/src/main/java/com/geek/firsthw/MainActivity.java
package com.geek.firsthw;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private Button button;
    private boolean newText;
    private TextView firstMsg;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
        setListenerOnBtn();
    }

    private void setListenerOnBtn() {
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(newText){
                    firstMsg.setText("Is True");
                    newText = false;
                }else {
                    firstMsg.setText("Is False");
                    newText = true;
                }
            }
        });
    }

    public void initViews(){
        firstMsg= findViewById(R.id.textView);
        button = findViewById(R.id.button);
    }


}

File: app/src/main/java/com/geek/myproject/ui/main/MainActivity.kt
package com.geek.myproject.ui.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.GridLayoutManager
import com.geek.myproject.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    lateinit var viewModel: MainViewModel
    val adapter: NotesRVAdapter = NotesRVAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        viewModel = ViewModelProviders.of(this)
            .get(MainViewModel::class.java)//достаем ViewModel(кэшируем вью модель)
        rv_notes.layoutManager = GridLayoutManager(this,2)
        rv_notes.adapter = adapter
        viewModel.viewState().observe(this, Observer { s->
            s?.let { adapter.notes = s.notes }})//S данные, которые приходят подписчикам от  вью модели
    }
}